pandas/core/frame.py

```diff
@@ -8419,7 +8419,9 @@ def _get_data(axis_matters):
                 raise NotImplementedError(msg)
             return data
 
-        if numeric_only is not None and axis in [0, 1]:
+        if (
+            self._is_homogeneous_type and self._mgr.any_extension_types and axis == 0
```

Comment:
do we really need all of these conditions?

Reply:
no I don't think so, as this is hiding some bugs (or inconsistencies)

```
>>> import pandas as pd
>>>
>>> pd.__version__
'1.1.0.dev0+1600.gdede2c7a1e'
>>>
>>> df = pd.DataFrame({"A": pd.Series([0, 1], dtype="category")})
>>> df
   A
0  0
1  1
>>>
>>> df.all()
A    False
dtype: bool
>>>
>>> df.A.all()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\pandas\pandas\core\generic.py", line 11394, in logical_func
    return self._reduce(
  File "C:\Users\simon\pandas\pandas\core\series.py", line 4028, in _reduce
    return delegate._reduce(name, skipna=skipna, **kwds)
  File "C:\Users\simon\pandas\pandas\core\arrays\categorical.py", line 2076, in _reduce
    raise TypeError(f"Categorical cannot perform the operation {name}")
TypeError: Categorical cannot perform the operation all
>>>
```

Reply:
will need to fix #21020 first